### PR TITLE
Make error:Cloneable public

### LINKED ISSUE #35182: error:Cloneable should be public
Please refer to: https://github.com/ballerina-platform/ballerina-spec/issues/820

### PR DESCRIPTION
**Pull request by KRVPerera**
## Purpose

Fixes #35182

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.error/src/main/ballerina/error.bal [hunk 1]
```diff
@@ -18,7 +18,7 @@ import ballerina/jballerina.java;
 
 # Type for value that can be cloned.
 # This is the same as in lang.value, but is copied here to avoid a dependency.
-type Cloneable readonly|xml|Cloneable[]|map<Cloneable>|table<map<Cloneable>>;
+public type Cloneable readonly|xml|Cloneable[]|map<Cloneable>|table<map<Cloneable>>;
```
Comment (gimantha) [inline langlib/lang.error/src/main/ballerina/error.bal:21]: Shall we also send a PR to the spec repo also?

**Review by gimantha — DISMISSED** [review]

File: tests/jballerina-unit-test/src/test/resources/test-src/error/error_test.bal [hunk 2]
```diff
@@ -199,7 +197,7 @@ function testGenericErrorWithDetailRecord() returns boolean {
     int detailStatusCode = 123;
     error e = error(reason, message = detailMessage, statusCode = detailStatusCode);
     string errReason = e.message();
-    map<value:Cloneable> errDetail = e.detail();
+    map<error:Cloneable> errDetail = e.detail();
```
Comment (hasithaa) [inline tests/jballerina-unit-test/src/test/resources/test-src/error/error_test.bal:200]: Do we have test cases, covering `value:Cloneable` ?

Comment (KRVPerera) [reply tests/jballerina-unit-test/src/test/resources/test-src/error/error_test.bal:200]: Cloneable is used in many other test areas and subsequent products of statements.

File: langlib/lang.error/src/main/ballerina/error.bal (hunk 1 above)
Comment (KRVPerera) [reply langlib/lang.error/src/main/ballerina/error.bal:21]: https://github.com/ballerina-platform/ballerina-spec/pull/1063

**Review by gimantha — APPROVED** [review]

**Comment by gimantha** [comment]
@hasithaa should we wait till https://github.com/ballerina-platform/ballerina-spec/pull/1063 get merged?

**Pull request merged**